### ISSUE TITLE
[TASK] Rename cache serializer to a common name

### ISSUE THREAD

**Issue by jlowe** [NVIDIA/spark-rapids#2975]
After #2905 all that remains is the Spark 3.1.1 cache serializer but it is common to all Spark versions that support a pluggable cache serializer.  This class should be renamed to a common package and moved out of the Spark 3.1.1 shim project.

**Comment by sperlingxx** [comment]
I am wondering how to adapt Spark 3.0.x if we move the serializer to common package. The `CachedBatchSerializer` was introduced at 3.1.

**Comment by jlowe** [comment]
The module would be common to all Spark versions >= 3.1.1 rather than in a specific shim directory as it is now.  That module would not compile against Spark 3.0.x.

**Comment by Salonijain27** [comment]
duplicate of #3314
